packages/mainmenu-extension/src/index.ts

```diff
@@ -397,10 +397,15 @@ export function createFileMenu(
           const apiURL = URLExt.join(setting.baseUrl, 'api/shutdown');
 
           // Shutdown all kernel and terminal sessions before shutting down the server
-          await Promise.all([
-            app.serviceManager.sessions.shutdownAll(),
-            app.serviceManager.terminals.shutdownAll()
-          ]);
+          // If this fails, we continue execution so we can post an api/shutdown request
+          try {
+            await Promise.all([
+              app.serviceManager.sessions.shutdownAll(),
+              app.serviceManager.terminals.shutdownAll()
+            ]);
+          } catch (e) {
+            // Do nothing
```

Comment:
I would print the error in the browser console to get it easily in case introspection is needed.

```suggestion
            // Do nothing
            console.log(`Failed to shutdown sessions and terminals: ${e}`);
```